packages/data-context/src/sources/migration/shouldShowSteps.ts

```diff
@@ -65,8 +65,14 @@ async function hasSpecFiles (projectRoot: string, dir: string, testFilesGlob: st
 export async function shouldShowAutoRenameStep (projectRoot: string, config: LegacyCypressConfigJson) {
   const specsToAutoMigrate = await getSpecs(projectRoot, config)
 
+  const e2eMigrationOptions = {
+    // If the configFile has projectId, we do not want to change the preExtension
+    // so, we can keep the cloud history
+    shouldMigratePreExtension: !config.projectId && !config.e2e?.projectId,
+  }
```

Comment:
Should this call this getter? https://github.com/cypress-io/cypress/pull/21193/files#diff-6dae8a6d4d0aac4cb2685a16ea20ceeb091f6113e7e882be5c14f89f579b7e52R69

Reply:
This file lives on the `data-context` folder but is not using the` dataContext` itself - I'll call this a util/helper file